test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,134 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alphabet/aminoacid/aa20.hpp>
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/search/algorithm/all.hpp>
+//
+using namespace seqan3;
+
+template<typename alphabet_t>
+void generate_text(std::vector<alphabet_t> & text,  unsigned const length)
+{
+    size_t const alphabet_size = alphabet_t::value_size; 
+    text.resize(length);
+    for (unsigned i = 0; i < length; ++i)
+    {
+        alphabet_t r;
+        r.assign_rank(std::rand() % alphabet_size);
+        text[i] = r;
+    }
+}
+
+template<typename alphabet_t>
+void mutate_insertion(std::vector<alphabet_t> & seq, size_t const  errors){
+    size_t const alphabet_size = alphabet_t::value_size;
+    size_t rpos = rand() % (seq.size() - errors);
+    size_t rrank = rand() % alphabet_size;
+    alphabet_t cbase;
+    cbase.assign_rank(rrank);
+    seq.insert(seq.begin() + rpos, cbase);
```

Comment:
```suggestion
    seq.insert(seq.begin() + rpos, alphabet_t{}.assign_rank(rrank));
```

Should work. Then you can remove the two lines above